### PR TITLE
1466415: Candlepin should allow setting of default auto-attach value …

### PR DESCRIPTION
**Pull request by wottop**
…for hypervisors

### CODE REVIEW

File: server/src/main/java/org/candlepin/controller/Entitler.java [hunk 1]
```diff
@@ -191,21 +199,35 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
...
     @Transactional
     public List<Entitlement> bindByProducts(AutobindData data, boolean force)
-        throws AutobindDisabledForOwnerException {
+        throws AutobindDisabledForOwnerException, AutobindHypervisorDisabledException {
         Consumer consumer = data.getConsumer();
         Owner owner = data.getOwner();
-
-        if ((!consumer.isDev() && owner.isAutobindDisabled()) || owner.isContentAccessEnabled()) {
+        ConsumerType type = this.consumerTypeCurator.getConsumerType(consumer);
+        if (!consumer.isDev() && (owner.isAutobindDisabled() ||
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/Entitler.java]: Is there any way we can pair down this beast of a conditional? Seems like we're doing a lot of very-specific context checking for this code.

If not, and if it's worth doing, we could add a simple utility function to ConsumerTypeEnum that could do the inverse check safely without needing to pretest for null:

```
public boolean isType(ConsumerType type) {
  return type != null && type.isType(this); // Alternatively, extend this out to not call into the type's isType method
}
```

File: server/src/main/java/org/candlepin/controller/Entitler.java [hunk 2]
```diff
@@ -191,21 +199,35 @@ public void adjustEntitlementQuantity(Consumer consumer, Entitlement ent,
...
+                " because of the hypervisor autobind setting" : "";
+            log.info("Skipping auto-attach for consumer '{}'. Auto-attach is disabled for owner {}{}{}",
+                consumer.getUuid(), owner.getKey(), caMessage, hypMessage);
+            if (!"".equals(hypMessage)) {
+                throw new AutobindHypervisorDisabledException(i18n.tr(
+                    "Auto-attach is disabled for owner \"{0}\"{1}.",
+                    owner.getKey(), hypMessage));
+            }
+            else {
+                throw new AutobindDisabledForOwnerException(i18n.tr(
```
Comment (Ceiu) [inline server/src/main/java/org/candlepin/controller/Entitler.java]: We can save a few lines of code here by using a tertiary op:

```throw new AutobindHypervisorDisabledException(..., !hypMessage.isEmpty() ? hypMessage : caMessage);```

Comment (wottop) [reply server/src/main/java/org/candlepin/controller/Entitler.java]: Those are different exceptions by design. Not sure we can use this. I'll change that condition.

Comment (Ceiu) [reply server/src/main/java/org/candlepin/controller/Entitler.java]: Ahh, my brain autocorrected the exception names. Disregard.

File: server/src/main/java/org/candlepin/controller/Entitler.java (hunk 1 above)
Comment (wottop) [reply server/src/main/java/org/candlepin/controller/Entitler.java]: I made it easier to read. Good enough?

Comment (Ceiu) [reply server/src/main/java/org/candlepin/controller/Entitler.java]: Yeah, that's much better. Thanks

**Pull request merged**